swarm/pss/pss.go

```diff
@@ -99,6 +99,7 @@ type Pss struct {
 	paddingByteSize int
 
 	// keys and peers
+	pubKeyPoolMu               sync.Mutex
```

Comment:
I was getting panics with `concurrent map read and write` on this map, when running tests with `exec` adapter.

Reply:
Do you know where? Ideally the lock already on the pss struct should take care of that.

Reply:
Yes, I noticed that we have a lock on the `pss` struct, but it is not enough apparently. I am not sure why it happens, but removing this particular lock and running the tests reproduces the panic consistently.

Reply:
@lmars would you happen to know about that lock issue? Am I using it wrong?

Reply:
@nolash `processAsym` is not locked with the `Pss.lock`. Not sure if it should be, but I believe the panics were happening on it.

Reply:
It most likely should, since it's accessing the asym peers map. Maybe replace this new mutex with the old lock and see if that works?

Reply:
I haven't looked at the code, but you should consider using `sync.Map`.

Reply:
@nolash adding the Pss lock works. I am sure I tried that, but probably the buffers were not amended at the time, and I thought this is not working for some reason. I tried with various number of nodes/messages/adapters and it seems to be stable.

Reply:
Great news. @lmars ' suggestion of `sync.Map` should be investigated probably, though. I'll have a look at that later on.